snuba/consumers/consumer_builder.py

```diff
@@ -91,3 +84,34 @@ def build_consumer(self) -> BatchingKafkaConsumer:
             queued_max_messages_kbytes=self.queued_max_messages_kbytes,
             queued_min_messages=self.queued_min_messages,
         )
+
+    def build_base_consumer(self) -> BatchingKafkaConsumer:
+        """
+        Builds the consumer with a ConsumerWorker.
+        """
+        return self.__build_consumer(
+            ConsumerWorker(
+                self.dataset,
+                producer=self.producer,
+                replacements_topic=self.replacements_topic,
+                metrics=self.metrics
+            )
+        )
+
+    def build_snapshot_aware_consumer(
+        self,
+        snapshot_id: SnapshotId,
+        transaction_data: TransactionData,
+    ) -> BatchingKafkaConsumer:
```

Comment:
Kind of wonder if it would make sense to use overloading here (we can type check it even with [the overload decorator](https://docs.python.org/3/library/typing.html#typing.overload)) or if that's getting too clever. I think it would read better.

Reply:
It's also a little weird that this will be exposed on non-CDC datasets. Can we isolate this better?

Reply:
about the overload. Actually I cannot because not only the parameter types are different. Their number is different as well. I would have to wrap them into a separate data structure which I think is defying the purpose.

Reply:
about exposing to non cdc datasets. In one of the previous implementation I tried to keep the logic that creates the worker out of the builder in a partially applied function, so that only the CDC specific states would pass that partially applied function. The result seemed overly complicated because the partially applied functions needed to access the content of the builder class anyway, since they need dataset, producer and metrics to instantiate the worker.